### PR TITLE
[Android] Support override back button press event inside navigation bar with FormsAppCompatActivity

### PR DESCRIPTION
**Pull request by Thetyne**
### Description of Change ###

Support override back button press event inside navigation bar material design with FormsAppCompatActivity

### Bugs Fixed ###

- Fail to migrate to android material design

### API Changes ###

List all API changes here (or just put None), example:

Added:
 - public delegate bool BackButtonPressedEventHandler(object sender, EventArgs e);
 - public static event BackButtonPressedEventHandler BackPressed;

Changed:
 - BarOnNavigationClick method

### CODE REVIEW

**Comment by ahmedalejo** [comment]
Who will subscribe to the event handler?

**Comment by Thetyne** [comment]
Inside mainActivity like the FormsAppCompatActivity
FormsAppCompatActivity.BackPressed += MyMethod;
Add this line
NavigationPageRenderer.BackPressed += MyMethod;

It's exactly the same code than FormsAppCompatActivity.


i think better trick must be reuse FormsAppCompatActivity.OnBackPressed() method who call FormsAppCompatActivity.BackPressed if exist.
But i don't know the better code for you.

Commented on [https://forums.xamarin.com/discussion/comment/202929#Comment_202929](url)

**Comment by jassmith** [comment]
Fix formatting then :+1:

**Comment by samhouts** [comment]
@Thetyne: Would you mind converting the spaces to tabs to match our style guide? Thank you!

**Comment by Thetyne** [comment]
Fixed

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 1]
```diff
@@ -28,7 +28,9 @@ namespace Xamarin.Forms.Platform.Android.AppCompat
 {
 	public class NavigationPageRenderer : VisualElementRenderer<NavigationPage>, IManageFragments
 	{
-		readonly List<Fragment> _fragmentStack = new List<Fragment>();
+		public delegate bool BackButtonPressedEventHandler(object sender, EventArgs e);
+
+        readonly List<Fragment> _fragmentStack = new List<Fragment>();
```
Comment (samhouts) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: So sorry to be a pain! This one is still spaces, though... Thank you!!!! :)

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 2]
```diff
@@ -357,12 +359,16 @@ void AnimateArrowOut()
 			valueAnim.Start();
 		}
 
-		void BarOnNavigationClick(object sender, AToolbar.NavigationClickEventArgs navigationClickEventArgs)
+        public static event BackButtonPressedEventHandler BackPressed;
```
Comment (brianlagunas) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: static events scare me. really need to test for memory leaks here.

**Comment by Thetyne** [comment]
I think that this fix corrected by a new code by pauldipietro #235.
Maybe close this issue after i test #235 patch

**Comment by Thetyne** [comment]
Ok, #235 is closed. i don't know reason.

I had fix formatting again

**Comment by Thetyne** [comment]
you can't close this fix, i used an other solution. 
view discussions : 
https://forums.xamarin.com/discussion/comment/218663#Comment_218663 
and
https://forums.xamarin.com/discussion/comment/218619#Comment_218619

**Pull request closed without merging**